### ISSUE TITLE
URL prefix not working

### ISSUE THREAD

**Issue by ytzelf** [pyload/pyload#2631]
I'm using nginx reverse proxy and setup'd pyload correctly running on port 546 with prefix /pyload.
However, assets (images, css and such) as well as web interface links are all absolute.

Use case
1 - I access https://my.server.557/pyload/login (very degraded, css less page) and login
2 - im redirected to http://myserver/ (no port, simple http, no prefix)
3 - i input https://my.server.557/pyload/queue and access a very degraded, css less queue page
4 - i click collector and am redirected to https://my.server.557/collector (no prefix) 

My nginx config is a simple 
       location ^~ /pyload/ {
                proxy_pass              http://127.0.0.1:546/pyload/;
        }

**Comment by wastez** [comment]
I use it like that for https:

```
location / {

        proxy_pass http://localhost:8000; # my existing pyload instance
        proxy_set_header Host $http_host; # $http_host uebergibt im gegensatz zu $host den port

        # re-write redirects to http as to https, example: /home
       proxy_redirect http:// https://;


```

**Comment by wastez** [comment]
As i can see this is wrong: "proxy_pass http://127.0.0.1:546/pyload/;"

If your instance is running on port 546 you have to look like this:
proxy_pass http://127.0.0.1:546;

**Comment by ytzelf** [comment]
Here are my headers at the server level
proxy_set_header                X-Real-IP       $remote_addr;
proxy_set_header                X-Forwarded-For $proxy_add_x_forwarded_for;

and location
location ^~ /pyload/ {
proxy_pass http://127.0.0.1:546;
proxy_set_header Host $http_host;
}

error does not change

**Comment by wastez** [comment]
Ok tested it and have a solution.

**Comment by ytzelf** [comment]
doesnt change a thing :( I'll keep digging

**Comment by wastez** [comment]
The /pyload is not so easy because the internal links of pyload shows to root, so it has to ne rewritten. 

Easier would be to use it just with the Port without /pyload. 
Just with the port (Location /), my First example is working.

**Comment by ytzelf** [comment]
yeah i know but that itches since every other app running is neatly attributed to relevant /xxxx ^^

**Comment by GammaC0de** [comment]
See #1128

**Comment by ytzelf** [comment]
thanks GammaC0de !
I saw that issue already and it does mention a workaround for apache, but I'm using nginx and don't understand why my simple proxy_pass doesn't work. 

I'll keep checking here if anyon with nginx succeded in making this work

**Comment by GammaC0de** [comment]
What about https://github.com/pyload/pyload/pull/2015?

**Comment by ytzelf** [comment]
I'm quite new to github; the changes referenced in https://github.com/pyload/pyload/pull/2015 are already comitted to the code? how do I implement them in my installatioN,

Sorry for that

**Comment by GammaC0de** [comment]
I'm afraid they don't

**Comment by ytzelf** [comment]
Well I tried downloading https://github.com/pyload/pyload/tree/2df92bf3c8b768cb62f2f2a3313064d25f37e760 and running the server based on that; this apparently contains the mentionned changes.
and the error is still here :(

**Comment by ytzelf** [comment]
Regarding label, this is more a bug than a question.. I checked home.html and links to other pages are hardcoded.

**Comment by GammaC0de** [comment]
@ytzelf I hove this would be fixed in the upcoming version, @vuolter can answer that.

**Comment by ytzelf** [comment]
fixing the html template is somewhat easy providing that there is an easy way to access settings in JS (and I think there is).
Actions fired through calls to "/json/xxxxx" are where I don't understand how url are formed.

**Comment by ytzelf** [comment]
I fixed the issue in a very dirty way with different variations of the command below in /module/web/templates/default/ and module/web/media/js/

sed -i 's/\/api/\/pyload\/api/g' module/web/templates/default/*

basically replacing /, /json/, /media/, and /api/ by /pyload, /json/pyload, /media/pyload, and /api/pyload

I feel dirty but it works

**Comment by youlJunior** [comment]
I use it with nginx, and no problems here.

My path prefix is "/pyload"

My Nginx conf is : 

location /pyload/ {
                auth_basic              'pyLoad';
                proxy_set_header        Host $host;
                proxy_set_header        X-Real-IP $remote_addr;
                proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header        X-Forwarded-Proto $scheme;
                proxy_pass              http://127.0.0.1:8000/;
                proxy_redirect          off;
        }

**Comment by RichardHeilmann** [comment]
Working version with [sub_filter](http://nginx.org/en/docs/http/ngx_http_sub_module.html) search and replace.
Beware:

>  This module is not built by default, it should be enabled with the --with-http_sub_module configuration parameter.

Fortunatelay on the nginx version on Ubuntu 16.04 it is.  
```
location /pyload/
	proxy_pass http://localhost:8000/;
	proxy_set_header Accept-Encoding "";
	sub_filter_types text/css text/xml text/javascript;
	sub_filter 'media/' 'pyload/media/';
	sub_filter 'json/' 'pyload/json/';
	sub_filter '<a href="/' '<a href="/pyload/';
	sub_filter_once off;
}
```

**Comment by 3mb3dw0rk5** [comment]
Needed to adapt the configuration of @RichardHeilmann due to api calls was not working on my nginx and the progress was not displayed on the home site.
This is what worked for me. Tested on Archlinux with nginx/1.10.3

`        location /pyload/ {
            proxy_pass http://localhost:8000/;
            proxy_set_header Accept-Encoding "";
            sub_filter_types text/css text/xml text/javascript;
            sub_filter '/media/' '/pyload/media/';
            sub_filter '/json/' '/pyload/json/';
            sub_filter '/api/' '/pyload/api/';
            sub_filter '<a href="/' '<a href="/pyload/';
            sub_filter_once off;
        }
`
